src/s3_log_extraction/summarize/_generate_all_dataset_summaries.py

```diff
@@ -0,0 +1,166 @@
+import collections
+import datetime
+import pathlib
+
+import pandas
+import tqdm
+
+from ..config import get_extraction_directory, get_summary_directory
+from ..ip_utils import load_ip_cache
+
+
+def generate_all_dataset_summaries() -> None:
```

Comment:
@rwblair An alternative idea I had here was to just name it the basic

```suggestion
def generate_summaries(level: int = 0) -> None:
```

but with the argument specifying which level (dataset being `level=0`) which is the 'top level' of the generic S3 bucket mirror, and the code could eventually (but fine to have `NotImplementedError` for now) support recursively _any_ level below that (e.g., per subject, per session, etc.)